libkbfs/tlf_handle.go

```diff
@@ -44,32 +44,39 @@ type TlfHandle struct {
 	name CanonicalTlfName
 }
 
-// IsPublic returns whether or not this TlfHandle represents a public
-// top-level folder.
-func (h TlfHandle) IsPublic() bool {
-	return h.public
+// Type returns the type of the TLF this TlfHandle represents.
+func (h TlfHandle) Type() tlf.Type {
+	return h.tlfType
 }
 
 // IsWriter returns whether or not the given user is a writer for the
 // top-level folder represented by this TlfHandle.
 func (h TlfHandle) IsWriter(user keybase1.UID) bool {
-	_, ok := h.resolvedWriters[user]
+	// TODO(KBFS-2185) relax this?
+	if h.tlfType == tlf.SingleTeam {
+		panic("Can't check whether a user is a writer on a team TLF")
```

Comment:
I don't understand why this statement is true. On the metadata structures we allow this check, why not on the handle?